x-pack/platform/packages/shared/kbn-ai-assistant/src/chat/welcome_message_knowledge_base.tsx

```diff
@@ -20,35 +20,60 @@ import {
 import usePrevious from 'react-use/lib/usePrevious';
 
 import { WelcomeMessageKnowledgeBaseSetupErrorPanel } from './welcome_message_knowledge_base_setup_error_panel';
-import type { UseKnowledgeBaseResult } from '../hooks/use_knowledge_base';
+import { UseKnowledgeBaseResult } from '../hooks';
 
 export function WelcomeMessageKnowledgeBase({
   knowledgeBase,
 }: {
   knowledgeBase: UseKnowledgeBaseResult;
 }) {
-  const prevIsInstalling = usePrevious(knowledgeBase.isInstalling);
+  const isPolling = !!knowledgeBase.status.value?.endpoint && !knowledgeBase.status.value?.ready;
+
+  // poll the status if isPolling (inference endpoint is created but deployment is not ready)
+  // stop when ready === true or some error
+  useEffect(() => {
+    if (!isPolling) {
+      return;
+    }
+
+    const interval = setInterval(async () => {
+      // re-fetch /status
+      await knowledgeBase.status.refresh();
+      const { value: currentStatus } = knowledgeBase.status;
+
+      // check if the model is now ready
+      if (currentStatus?.ready) {
+        // done installing
+        clearInterval(interval);
+        return;
+      }
+    }, 5000);
```

Comment:
Only `refresh()` should be invoked every interval. I think the other check can be moved outside

```suggestion
    const interval = setInterval(async () => {
      statusRequest.refresh();
    }, 5000);
```